tiles/push_block.gd

```diff
@@ -0,0 +1,40 @@
+extends KinematicBody2D
+
+class_name PushBlock,"res://tiles/push_block.png"
+
+onready var tween := $Tween
+onready var ray := $RayCast2D
+
+var push_acumulator = 0
+
+func _process(delta):
+	if push_acumulator>0:
+		push_acumulator -= delta
+	if push_acumulator < 0:
+		push_acumulator == 0
+
+
+func push(delta,from:Vector2):
+	var relative_pos = (global_position - from).snapped(Vector2(16,16))
+	ray.cast_to = relative_pos.normalized()*16
+	push_acumulator = push_acumulator+2*delta
+	if push_acumulator<1: return
+	if tween.is_active(): return
+	if ray.get_collider(): return
```

Comment:
I feel like this section could use some refactoring, it's a simple class but we can extract some simple logic out of this function into smaller functions

These three if statements can be extracted into a function. Something along the lines of short_circuit_push(). If that function returns true then return. The whole functions reads faster even though logic is no being separated.

Reply:
This section is a series of "bail out" or "guards". It's an idiom in computing.

Reply:
perfect, idioms make naming easier, see below